### PR TITLE
Better handle errors in entry points

### PR DESCRIPTION
**Pull request by AstraLuma**
### What does this PR do?

Fixes entry point loading, so that if it fails, the exception is logged instead of propagated to the larger system. 

### What issues does this PR fix or reference?

None.

### Previous Behavior
The minion would fail to start if there was any kind of problem with entry point loading.

### New Behavior
The exception is logged instead.

### Tests written?

No, and I'm not sure how or where.

### Commits signed with GPG?

Yes.

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by cachedout** [comment]
Is there a bug filed against this issue? I have never heard of this failing before and I am curious as to why it would do so. I'd love to see the exact stacktrace here.

**Comment by astronouth7303** [comment]
I found problems where if you have a bad entry point declared (in my case, because of copy/paste errors and then the code wasn't being packaged), then salt won't start. There's also weird things that happen if there's a problem in the called function.

You probably hadn't heard of anyone encountering this problem because this feature is completely undocumented (see #50633) and nobody has used it as far as I know. (Except now, because this is super helpful for my own projects.)

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by cachedout** [comment]
@astronouth7303 Thanks for the explanation. This is a reasonably safe change for us so we're happy to get this in but I do think you should make the changes that @isbm recommends before we merge it.

File: salt/loader.py [hunk 1]
```diff
@@ -160,9 +160,12 @@ def _module_dirs(
...
             for entry_point in pkg_resources.iter_entry_points('salt.loader', ext_type_dirs):
-                loaded_entry_point = entry_point.load()
-                for path in loaded_entry_point():
-                    ext_type_types.append(path)
+                try:
+                    loaded_entry_point = entry_point.load()
+                    for path in loaded_entry_point():
+                        ext_type_types.append(path)
+                except Exception:
+                    log.exception("Error running entry point %r", entry_point)
```
[the remark this answers is not in the document]
Comment (astronouth7303) [reply salt/loader.py]: The developer would certainly strongly appreciate the traceback, and I would consider any time this code is triggered to be a bug in some third-party code.

**Comment by astronouth7303** [comment]
Ok, logging is improved.

File: salt/loader.py [hunk 2]
```diff
@@ -138,6 +138,18 @@ def static_loader(
     return ret
 
 
+def _format_entrypoint_target(ep):
+    """
```
Comment (cachedout) [inline salt/loader.py]: The Salt style guide indicates that these should be three single-quotes.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Review by isbm — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Pull request merged**